File: captcha/verify.js
const PADDING = 15

export function verify(clicks, positions) {
  if (clicks.length !== positions.length) {
    return false
  }
  for (let i = 0; i < clicks.length; i++) {
    const [cx, cy] = clicks[i]
    const [px, py, sz] = positions[i]
    if (
      cx < px - PADDING ||
      cx > px + sz + PADDING ||
      cy < py - PADDING ||
      cy > py + sz + PADDING
    ) {
      return false
    }
  }
  return true
}

File: svg_captcha/src/verify.rs
/// Click tolerance in pixels.
///
/// 点击容差（像素）。
const PADDING: i32 = 15;

/// Verifies if the provided clicks match the positions of the target icons.
///
/// 验证提供的点击是否匹配目标图标的位置。
pub fn verify(clicks: &[(i32, i32)], positions: &[(i32, i32, u32)]) -> bool {
    clicks.len() == positions.len()
        && clicks
            .iter()
            .zip(positions)
            .all(|(&(cx, cy), &(px, py, sz))| {
                let sz = sz as i32;
                cx >= px - PADDING
                    && cx <= px + sz + PADDING
                    && cy >= py - PADDING
                    && cy <= py + sz + PADDING
            })
}
